Garrett-Smith10/recipeproject_client
# feature/login_and_register

Description
This pull request introduces login and registration functionality to the frontend of the application. The following changes have been made:

Login Page: Users can now log in using their username and password. Validation errors are displayed if the credentials are incorrect.
Registration Page: Users can register by providing a username, email, and password. Validation checks ensure that the input meets the required criteria.
Authentication Flow: Upon successful login or registration, users are redirected to their dashboard/home page.
Changes
Components Added:

Login.jsx: Component for user login.
Register.jsx: Component for user registration.
Updated Components:

NavBar.jsx: Navigation updated to handle authentication states.
Routes Added:

/login: Route for the login page.
/register: Route for the registration page.
How to Test
Login:

Navigate to /login.
Enter valid credentials to log in.
Ensure that you are redirected to the dashboard/home page.
Registration:

Navigate to /register.
Fill out the registration form and submit.
Verify that a new user is created and redirected to the login page or dashboard as specified.
